src/dp.rs

```diff
@@ -19,50 +62,66 @@ pub trait DifferentialPrivacyBudget {}
 /// Marker trait for differential privacy scalar noise distributions.
 pub trait DifferentialPrivacyDistribution {}
 
-/// Zero-concentrated differential privacy (zCDP) budget as defined in [[BS16]].
+/// Zero-concentrated differential privacy (ZCDP) budget as defined in [[BS16]].
 ///
 /// [BS16]: https://arxiv.org/pdf/1605.02065.pdf
 pub struct ZeroConcentratedDifferentialPrivacyBudget {
-    /// Parameter `epsilon`, using the notation from [[CKS20]] where `rho = (epsilon**2)/2`
-    /// for a `rho`-zCDP budget.
-    ///
-    /// [CKS20]: https://arxiv.org/pdf/2004.00010.pdf
-    pub epsilon: Rational,
+    epsilon: BigURational,
 }
 
-impl DifferentialPrivacyBudget for ZeroConcentratedDifferentialPrivacyBudget {}
+/// Alias for ZeroConcentratedDifferentialPrivacyBudget.
+pub type ZCdpBudget = ZeroConcentratedDifferentialPrivacyBudget;
 
-/// Zero-mean Discrete Gaussian noise distribution.
-///
-/// The distribution is defined over the integers, represented by arbitrary-precision integers.
-#[derive(Clone, Debug)]
-pub struct DiscreteGaussian {
-    /// Standard deviation of the distribution.
-    pub sigma: Rational,
-}
+impl ZCdpBudget {
+    /// Create a budget for parameter `epsilon`, using the notation from [[CKS20]] where `rho = (epsilon**2)/2`
+    /// for a `rho`-ZCDP budget.
+    ///
+    /// [CKS20]: https://arxiv.org/pdf/2004.00010.pdf
+    pub fn new(epsilon: Rational) -> Self {
+        ZCdpBudget {
+            epsilon: epsilon.into(),
+        }
+    }
 
-impl DifferentialPrivacyDistribution for DiscreteGaussian {}
+    /// Create a budget for parameter `epsilon`, using the notation from [[CKS20]] where `rho = (epsilon**2)/2`
+    /// for a `rho`-ZCDP budget. Returns a `DPError` if the input float is not finite and positive.
+    ///
+    /// [CKS20]: https://arxiv.org/pdf/2004.00010.pdf
+    pub fn from_float(epsilon: f32) -> Result<Self, DPError> {
```

Comment:
i'm a bit confused. `Rational` is only intended to avoid exposing `Ratio<BigUint>` to the public API, is it? i could implement `TryFrom<f32>` for `Ratio<BigUint>` and use that in this `ZCdpBudget` constructor? why do we want to convert a `Rational` to a `DiscreteGaussian`, and how?

Reply:
My apologies, I misread the diff. I meant `ZCdpBudget`, not `DiscreteGaussian`.

Let me try again:

Here we're taking a `f32` and constructing a `ZCdpBudget`. What we could do is instead is write a method to construct a `Rational` from an `f32` so that we don't need the extra constructor.

The advantage is that, if we have other situations like this, we can reuse the `f32` -> `Rational` constructor.